lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/driver/YarnDriverRestartConfiguration.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.runtime.yarn.driver;
+
+import org.apache.reef.annotations.Provided;
+import org.apache.reef.annotations.audience.ClientSide;
+import org.apache.reef.annotations.audience.Public;
+import org.apache.reef.client.DriverRestartConfiguration;
+import org.apache.reef.driver.parameters.DriverRestartHandler;
+import org.apache.reef.tang.formats.ConfigurationModule;
+import org.apache.reef.tang.formats.ConfigurationModuleBuilder;
+import org.apache.reef.tang.formats.RequiredImpl;
+import org.apache.reef.wake.EventHandler;
+import org.apache.reef.wake.time.event.StartTime;
+
+/**
+ * Use this ConfigurationModule to configure YARN-specific Restart options for the driver.
+ * Note that there is a required parameter ON_DRIVER_RESTARTED in DriverRestartConfiguration to bind.
+ * <p/>
+ */
+@ClientSide
+@Public
+@Provided
+public final class YarnDriverRestartConfiguration extends ConfigurationModuleBuilder {
+  /**
+   * This event is fired in place of the ON_DRIVER_STARTED when the Driver is in fact restarted after failure.
+   */
+  public static final RequiredImpl<EventHandler<StartTime>> ON_DRIVER_RESTARTED = new RequiredImpl<>();
```

Comment:
My original intention here is to force the user to bind at least one RestartHandler, since in my first PR for REEF-483, I was using the existence of a RestartHandler to determine whether or not the user wants to restart their job. On second thought, I should probably be using the existence of the runtime RestartManager (to be implemented in REEF-483) to determine that instead, Will remove.